src/data-access/user.ts

```diff
@@ -0,0 +1,83 @@
+import { Op } from 'sequelize';
+import { UserModel, initUserModel } from '../models/user';
+
+import { User } from '../types/user';
+
+function convertToUser(model: UserModel): User | null {
+  if (!model) {
+    return null;
+  }
+
+  return {
+    id: model.id,
+    login: model.login,
+    password: model.password,
+    age: model.age,
+    isDeleted: model.isDeleted
+  };
+}
+
+export interface IAccessorToUserData<T> {
+  getById(id: string): Promise<T | null>;
+  createOrUpdate(user: User): Promise<User | null>;
+  getSomeBySubstring(regexpString: string, limit?: number | undefined): Promise<User[] | null>;
+}
+
+export class AccessorToUserData implements IAccessorToUserData<User> {
```

Comment:
Class name have to speak by itself. In this case it is very unspecified what this class has to do

Reply:
Done